text/0000-named-tracers.md

```diff
@@ -0,0 +1,108 @@
+
+On an SDK level, the SpanData class and its Metrics counterpart are extended with a `getLibraryResource` function that returns the resource associated with the Tracer / Meter that created it.
+
+If there are two different instrumentation libraries for the same technology (e.g. MongoDb), these instrumentation libraries should have distinct names.
+
+## Prior art and alternatives
+
+This proposal originates from an `opentelemetry-specification` proposal on [components](https://github.com/open-telemetry/opentelemetry-specification/issues/10) since having a concept of named Tracers would automatically enable determining this semantic `component` property.
+
+Alternatively, instead of having a `TracerFactory`, existing (global) Tracers could return additional indirection objects (called e.g. `TraceComponent`), which would be able to produce spans for specifically named traced components.
+
+```java
+TraceComponent traceComponent = OpenTelemetry.Tracing.getTracer().componentBuilder(libraryResource);
+Span span = traceComponent.spanBuilder("someMethod").startSpan();
+```
+
+Overall, this would not change a lot compared to the `TracerFactory` since the levels of indirection until producing an actual span are the same.
+
+Instead of setting the `component` property based on the given Tracer names, those names could also be used as *prefixes* for produced span names (e.g. `<TracerName-SpanName>`). However, with regard to data quality and semantic conventions, a dedicated `component` set on spans is probably preferred.
+
+Instead of using plain strings as an argument for creating new Tracers, a `Resource` identifying an instrumentation library could be used. Such resources must have a _version_ and a _name_ label (there could be semantic convention definitions for those labels). This implementation alternative mainly depends on the availability of the `Resource` data type on an API level (see https://github.com/open-telemetry/opentelemetry-specification/pull/254).
+
+```java
+// Create resource for given instrumentation library information (name + version)
+Map<String, String> libraryLabels = new HashMap<>();
+libraryLabels.put("name", "io.opentelemetry.contrib.mongodb");
+libraryLabels.put("version", "1.0.0");
+Resource libraryResource = Resource.create(libraryLabels);
+// Create tracer for given instrumentation library.
+Tracer tracer = OpenTelemetry.getTracerFactory().getTracer(libraryResource);
+```
+
+Those given alternatives could be applied to Meters and Metrics in the same way.
+
+## Future possibilities
+
+Based on the Resource information identifying a Tracer or Meter these could be configured (enabled / disabled) programmatically or via external configuration sources (e.g. environment).
+
+Based on this proposal, future "signal producers" (i.e. logs) can use the same or a similar creation approach.
+
+## Examples (of Tracer and Meter names)
```

Comment:
Maybe this section should be moved to a subsection of Explanation or Internal Details, so that we don't have to point to "see examples at the end" there.